mm2src/coins/coin_errors.rs

```diff
@@ -0,0 +1,80 @@
+use crate::{utxo::rpc_clients::UtxoRpcError, DelegationError, NumConversError, TxHistoryError,
+            UnexpectedDerivationMethod, WithdrawError};
+use futures01::Future;
+use mm2_err_handle::prelude::MmError;
+use spv_validation::helpers_validation::SPVError;
+
+pub type ValidatePaymentFut<T> = Box<dyn Future<Item = T, Error = MmError<ValidatePaymentError>> + Send>;
+
+#[derive(Debug, Display)]
+pub enum ValidatePaymentError {
+    InternalError(String),
+    InvalidPaymentTxData(String),
+    InvalidInput(String),
+    InvalidRpcResponse(String),
+    SPVError(SPVError),
+    UnexpectedPaymentState(String),
+    Transport(String),
+    WrongPaymentTx(String),
```

Comment:
What is the difference between `WrongPaymentTx` and `InvalidPaymentTxData`? It seems that one of them can be removed.

Reply:
I guess so as the differences are interchangeable between the two error kind so I don't think they have any unique differences and we can simply use `WrongPaymentTxData` instead of the these.

what do you think @sergeyboyko0791?

Reply:
> InvalidPaymentTxData means that there was a problem with deserializing the transaction, or one of the transaction parts is invalid.
WrongPaymentTx means that the transaction has wrong properties, for example, it has been sent to a wrong address.

Isn't `one of the transaction parts is invalid.` and `the transaction has wrong properties` actually the same? For more clarity, `InvalidPaymentTxData` should be changed to `TxDeserializationError`.